pycbc/io/live.py

```diff
@@ -388,18 +388,89 @@ def upload(self, fname, gracedb_server=None, testing=True,
                 self.gracedb.write_label(gid, 'INJ')
                 logging.info("Tagging event %s as an injection", gid)
 
-            # add info for tracking code version
-            gracedb_tag_with_version(self.gracedb, gid)
-
-            extra_strings = [] if extra_strings is None else extra_strings
-            for text in extra_strings:
-                self.gracedb.write_log(gid, text, tag_name=['analyst_comments'])
         except Exception as exc:
-            logging.error('Something failed during the upload/annotation of '
+            logging.error('Something failed during the upload of '
                           'event %s on GraceDB. The event may not have been '
                           'uploaded!', fname)
             logging.error(str(exc))
 
+        # Upload em_bright properties JSON
+        if self.hasmassgap is not None:
+            try:
+                self.gracedb.write_log(
+                    gid, 'EM Bright properties JSON file upload',
+                    filename=self.embright_file,
+                    tag_name=['em_bright']
+                )
+                logging.info('Uploaded em_bright properties for %s', gid)
+            except Exception as exc:
+                logging.error('Failed to upload em_bright properties file '
+                              'for %s', gid)
+                logging.error(str(exc))
+
+        # Upload multi-cpt p_astro JSON
+        if self.astro_probs is not None:
+            try:
+                self.gracedb.write_log(
+                    gid, 'Multi-component p_astro JSON file upload',
+                    filename=self.multipa_file,
+                    tag_name=['p_astro'],
+                    label='PASTRO_READY'
+                )
+                logging.info('Uploaded multi p_astro for %s', gid)
+            except Exception as exc:
+                logging.error('Failed to upload multi p_astro file for %s', gid)
+                logging.error(str(exc))
+
+        # If there is p_astro but no probabilities, upload p_astro JSON
+        if hasattr(self, 'pastro_file'):
+            try:
+                self.gracedb.write_log(
+                    gid, '2-component p_astro JSON file upload',
+                    filename=self.pastro_file,
+                    tag_name=['sig_info']
+                )
+                logging.info('Uploaded p_astro for %s', gid)
+            except Exception as exc:
+                logging.error('Failed to upload p_astro file for %s', gid)
+                logging.error(str(exc))
+
+        # If there is no multi p_astro, upload source probabilities in JSON
```

Comment:
So following Tito's latest comment this block of code should be moved to after the block dealing with SNR timeseries and PSD plots but just before the 'extra strings' block.  Also remove the 'if self.astro_probs is None' switch, ie if self.prob_file exists then the plot should always be made and uploaded.